src/platform/packages/shared/kbn-workflows-ui/src/components/workflow_selector/workflow_selector.tsx

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import {
+  EuiFormRow,
+  EuiHighlight,
+  EuiIcon,
+  EuiInputPopover,
+  EuiLink,
+  EuiLoadingSpinner,
+  EuiPopoverFooter,
+  EuiSelectable,
+  type EuiSelectableOption,
+  EuiText,
+  EuiToolTip,
+  useEuiTheme,
+} from '@elastic/eui';
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { useWorkflows } from '../../hooks';
+import * as i18n from './translations';
+import { WorkflowSelectorEmptyState } from './workflow_selector_empty_state';
+import { getSelectedWorkflowDisabledError, processWorkflowsToOptions } from './workflow_utils';
+import type { WorkflowOption, WorkflowSelectorConfig } from './workflow_utils';
+import { IconDisabledWorkflow } from '../../assets/icons';
+
+interface WorkflowSelectorProps {
+  selectedWorkflowId?: string;
+  onWorkflowChange: (workflowId: string) => void;
+  config?: WorkflowSelectorConfig;
+  error?: string;
+  onCreateWorkflow?: () => void;
```

Comment:
This prop does not seem necessary. Are we doing something different from outside?